blue_bringup/scripts/calibrate_joint_offsets_new.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+
+import numpy as np
+
+
+def old(actuator_angles):
+
+    side = "right"
+    version = "2"
+
+    gr1 = 7.1875
+    gr2 = 8.2444852941
+
+    if side == 'left':
+        rot2 = np.pi / 2
+    elif side == 'right':
+        rot2 = -np.pi / 2
+
+    if version == '1':
+        rot1 = -2.189
+    elif version == '2':
+        rot1 = -2.310
+
+    ms = actuator_angles
+    a0 = ms[0]
+    a1 = ms[1] + rot1 * gr1 + rot2 * gr2
+    a2 = ms[2] + -rot1 * gr1 + rot2 * gr2
+    a3 = ms[3] + rot1 * gr1 - rot2 * gr2
+    a4 = ms[4] + -rot1 * gr1 - rot2 * gr2
+    a5 = ms[5] + -2.189 * gr1
+    a6 = ms[6] + 2.189 * gr1
+    actuators = [a0, a1, a2, a3, a4, a5, a6]
+
+    return np.array(actuators)
+
+
+def new(actuator_angles):
+
+    # Helper function for propagating joint angles => actuator angles
+    def actuator_angles_from_joint_angles(joint_angles):
+
+        # Transmission helpers
+        def propagate_through_base_roll(base_roll_angle):
+            base_roll_ratio = 7.1875
+            return base_roll_ratio * base_roll_angle
+
+        def propagate_through_differential(lift_angle, roll_angle):
+            lift_ratio = 7.1875
+            roll_ratio = 8.2444852941
+            right_actuator_angle = lift_angle * lift_ratio + roll_angle * roll_ratio
+            left_actuator_angle = -lift_angle * lift_ratio + roll_angle * roll_ratio
+            return (right_actuator_angle, left_actuator_angle)
```

Comment:
Maybe rename right to left and left to right? (This result is correct but could be confusing since now the left side motor is the first one in the robot chain)